src/components/Courses/Courses.jsx

```diff
@@ -1,117 +1,79 @@
 import { useState } from 'react';
-import { v4 as uuidv4 } from 'uuid';
+import { Link } from 'react-router-dom';
+import { PropTypes } from 'prop-types';
+
 import CourseCard from './components/CourseCard/CourseCard';
 import { mockedCoursesList, mockedAuthorsList } from '../../constants';
 import addAuthors from '../../helpers/addAuthors';
 import pipeDuration from '../../helpers/pipeDuration';
 import dateGenerator from '../../helpers/dateGenerator';
 import SearchBar from './components/SearchBar/SearchBar';
 import Button from '../../common/Button/Button';
-import CreateCourse from '../CreateCourse/CreateCourse';
 import './courses.css';
 
-function Courses() {
+function Courses({ courses, setCourses }) {
 	const [query, setQuery] = useState('');
-	const [courses, setCourses] = useState(mockedCoursesList);
-	const [newCourseForm, setNewCourseForm] = useState(false);
-	const [courseData, setCourseData] = useState({
-		title: '',
-		description: '',
-		authors: [],
-		duration: '',
-	});
-	const courseList = courses.map((item, i) => {
+
+	const renderCorsecard = (item, i) => {
 		return (
 			<CourseCard
 				key={i}
 				title={item.title}
+				courseID={item.id}
 				description={item.description}
-				authors={addAuthors(item.authors, mockedAuthorsList)}
+				authors={addAuthors(item.authors, mockedAuthorsList).join(', ')}
 				duration={pipeDuration(item.duration)}
 				creationDate={dateGenerator(item.creationDate)}
 			/>
 		);
-	});
+	};
+
+	const courseList = courses.map((item, i) => renderCorsecard(item, i));
```

Comment:
I don't think it was necessary to do it like this. Your previous approach was ok.